### PR TITLE
Banish stdin/stdout for ledger

### PR DESCRIPTION
**Pull request by rob-solana**
step one: accept a "ledger file" argument instead of "outfile"

### CODE REVIEW

File: src/fullnode.rs [hunk 1]
```diff
@@ -31,13 +31,8 @@ pub struct FullNode {
...
 }
 
-pub enum InFile {
-    StdIn,
-    Path(String),
-}
-
-pub enum OutFile {
-    StdOut,
+pub enum Ledger {
```
Comment (garious) [inline src/fullnode.rs]: Needs a more precise name

File: src/fullnode.rs [hunk 2]
```diff
@@ -31,13 +31,8 @@ pub struct FullNode {
...
+    StdInOut,
```
Comment (garious) [inline src/fullnode.rs]: Why support this?

Comment (rob-solana) [reply src/fullnode.rs]: migration

File: src/fullnode.rs (hunk 1 above)
Comment (rob-solana) [reply src/fullnode.rs]: any suggestions?

**Comment by rob-solana** [comment]
I think this is ready for review

File: multinode-demo/setup.sh [hunk 3]
```diff
@@ -86,8 +86,8 @@ if $node_type_leader; then
   echo "Creating $SOLANA_CONFIG_DIR/mint.json with $num_tokens tokens"
   $solana_mint <<<"$num_tokens" > "$SOLANA_CONFIG_PRIVATE_DIR"/mint.json
 
-  echo "Creating $SOLANA_CONFIG_DIR/genesis.log"
-  $solana_genesis < "$SOLANA_CONFIG_PRIVATE_DIR"/mint.json > "$SOLANA_CONFIG_DIR"/genesis.log
+  echo "Creating $SOLANA_CONFIG_DIR/ledger.log"
+  $solana_genesis < "$SOLANA_CONFIG_PRIVATE_DIR"/mint.json > "$SOLANA_CONFIG_DIR"/ledger.log
```
Comment (garious) [inline multinode-demo/setup.sh]: That file is the genesis block. Why rename to something less precise?

Comment (rob-solana) [reply multinode-demo/setup.sh]: ledger.log will be re-used forever, "genesis" seemed inappropriate.

once the network is up, is separation of the genesis block important?

I can add a step in bash to copy genesis.log into ledger.log, but genesis.log becomes an appendix, a tonsil

**Comment by garious** [comment]
@rob-solana, that merge commit I added from the web ui isn't going to work out because we disabled doing a 2nd merge commit on the PR itself. Can you rebase and `push -f` so we can get this merged?

**Pull request merged**